Realm/RLMSyncManager.mm

```diff
@@ -111,6 +115,7 @@ - (instancetype)initWithCustomRootDirectory:(NSURL *)rootDirectory {
         auto mode = should_encrypt ? SyncManager::MetadataMode::Encryption : SyncManager::MetadataMode::NoEncryption;
         rootDirectory = rootDirectory ?: [NSURL fileURLWithPath:RLMDefaultDirectoryForBundleIdentifier(nil)];
         SyncManager::shared().configure(rootDirectory.path.UTF8String, mode, "", none, true);
```

Comment:
Minor, but the ObjectStore API is designed so the `""` here should have been the binding part since it never changes. Object Store then combines the binding and application parts.

Instead, you construct binding+application part in the `userAgent` method.

Reply:
The reason for splitting them is that we want to send the binding part irrespectively of the user setting the application part.